code/modules/projectiles/ammo_types/xenos/pyrogen_xenoammo.dm

```diff
@@ -25,5 +25,12 @@
 	new /obj/effect/temp_visual/xeno_fireball_explosion(get_turf(target_atom))
 	for(var/turf/affecting AS in RANGE_TURFS(1, target_atom))
 		new /obj/fire/melting_fire(affecting)
-		for(var/mob/living/carbon/fired in affecting)
-			fired.take_overall_damage(PYROGEN_FIREBALL_AOE_DAMAGE, BURN, FIRE, FALSE, FALSE, TRUE, 0, , max_limbs = 2)
+		for(var/atom/fired in affecting)
```

Comment:
so the contents of a tile MUST be atom/movable. So instead of doing atom in (which does an istype check on every single thing but will never fail) we can do below.
```suggestion
		for(var/atom/movable/fired AS in affecting)
```